src/components/ucp-tabs/content-manager/buttons/install-button.tsx

```diff
@@ -0,0 +1,128 @@
+import { useAtomValue, useSetAtom } from 'jotai';
+import { Download } from 'react-bootstrap-icons';
+import { STATUS_BAR_MESSAGE_ATOM } from '../../../footer/footer';
+import {
+  BUSY_CONTENT_ELEMENTS_ATOM,
+  CONTENT_ELEMENTS_ATOM,
+  CONTENT_INTERFACE_STATE_ATOM,
+  CONTENT_TAB_LOCK,
+  EXTENSIONS_STATE_IS_DISK_DIRTY_ATOM,
+} from '../state/atoms';
+import { downloadContent } from './callbacks/download-and-install-content';
+import { CONTENT_TREE_ATOM } from './callbacks/dependency-management';
+import { showModalOk } from '../../../modals/modal-ok';
+import { getStore } from '../../../../hooks/jotai/base';
+
+// eslint-disable-next-line import/prefer-default-export
+export function InstallButton(
+  props: React.ButtonHTMLAttributes<HTMLButtonElement>,
+) {
+  const setStatusBarMessage = useSetAtom(STATUS_BAR_MESSAGE_ATOM);
+
+  const interfaceState = useAtomValue(CONTENT_INTERFACE_STATE_ATOM);
+
+  const contentElements = useAtomValue(CONTENT_ELEMENTS_ATOM);
+
+  const selectionCount = interfaceState.selected.length;
+
+  const countElement =
+    selectionCount < 2 ? (
+      <span />
+    ) : (
+      <span className="m-auto">( {selectionCount} )</span>
+    );
+
+  const installedCount = interfaceState.selected.filter(
+    (ce) => ce.installed,
+  ).length;
+  const onlineOnlyCount = interfaceState.selected.filter(
+    (ce) => !ce.installed && ce.online,
+  ).length;
+  const busyCount = useAtomValue(BUSY_CONTENT_ELEMENTS_ATOM).length;
+
+  let enabled = true;
+  let helpText = 'Install selected content';
+  if (installedCount > 0) {
+    enabled = false;
+    helpText = 'Selection includes already installed content';
+  } else if (onlineOnlyCount === 0) {
+    enabled = false;
+    helpText = 'Select at least one content element';
+  } else if (busyCount > 0) {
+    enabled = false;
+    helpText =
+      'Cannot install content when other content is processing. Please wait first';
+  }
+
+  const tree = useAtomValue(CONTENT_TREE_ATOM);
+
+  return (
+    <div
+      onMouseEnter={() => {
+        setStatusBarMessage(helpText);
+      }}
+      onMouseLeave={() => {
+        setStatusBarMessage(undefined);
+      }}
+    >
+      <button
+        className="ucp-button ucp-button-variant"
+        disabled={tree === undefined || !enabled}
+        type="button"
+        onClick={async () => {
+          getStore().set(
+            CONTENT_TAB_LOCK,
+            getStore().get(CONTENT_TAB_LOCK) + 1,
+          );
+
+          const ids = interfaceState.selected.map(
+            (ec) => `${ec.definition.name}@${ec.definition.version}`,
+          );
+
+          const solution = tree!.dependenciesForMultiple(ids);
+
+          if (solution.status !== 'OK') {
+            await showModalOk({
+              title: 'Cannot install',
+              message: solution.message,
```

Comment:
So, next ugly question: If these are modals and therefore presented to the user. Which parts need localization?

Reply:
Was this resolved with the review comments?